src/layers/graphlayers.jl

```diff
@@ -0,0 +1,18 @@
+abstract type AbstractGraphLayer end
+
+(l::AbstractGraphLayer)(x::AbstractMatrix) = l(l.fg, x)
+
+
+abstract type NodeFeatureLayer <: AbstractGraphLayer end
+
+(l::NodeFeatureLayer)(fg::FeaturedGraph) = FeaturedGraph(fg.graph, nf = l(fg, node_feature(fg)))
+
+
+abstract type EdgeFeatureLayer <: AbstractGraphLayer end
+
+(l::EdgeFeatureLayer)(fg::FeaturedGraph) = FeaturedGraph(fg.graph, ef = l(fg, edge_feature(fg)))
+
+
+abstract type GraphFeatureLayer <: AbstractGraphLayer end
+
+(l::GraphFeatureLayer)(fg::FeaturedGraph) = FeaturedGraph(fg.graph, gf = l(fg, graph_feature(fg)))
```

Comment:
maybe this is too much type hierarchy, and also doesn't cover all possibilities since layers could update both node and edge features (although in practice almost all layers would be of the NodeFeatureLayer type).
I think we can just have 
```
(l::AbstractGraphLayer)(fg::FeaturedGraph) = FeaturedGraph(fg.graph, nf = l(fg, node_feature(fg)))
```
as a generic default

Reply:
Well...............yeah, most of graph convolutional layer performs operations on node feature. However, there still some graph convolutional layer is performed on edge feature or global feature, which is currently not implemented in GeometricFlux. That's why I am struggling putting this API on `AbstractGraphLayer`.

Reply:
Layer that also operate on edge features will define their specializations

```julia
function (l::NewLayer)(fg::FeaturedGraph)
    V, E = l(fg, node_feature(fg), edge_feature(E)) 
    FeaturedGraph(fg.graph, nf = V, ef = E)
end
```

Reply:
Sure. I know that could be defined. But I don't know what decision will make in this scenario. One option is to document it and ask users to define for their own cases. Or we establish the type hierarchy and define the behavior for them. If too nested type hierarchy is to prevent, it is tend to the first option. To my knowledge, almost GNN layers only operate single feature (node/edge/global). There is very few cases operate two features or above.

Reply:
I think it is better to keep a simpler hierarchy, it's already difficult to figure out what is going on with `MessagePassing` and `GraphNet`

We should create a "Create custom layers" section in the docs with two fully worked-out examples, one updating only node features, the with nodes and edges updates

Reply:
Great!